src/net_processing.cpp

```diff
@@ -4200,6 +4209,53 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         peer->m_recon_state->m_next_recon_respond = NextReconRespond(current_time);
         return;
     }
+
+    // Received a response to the reconciliation request.
+    // May leak tx-related privacy if we announce local transactions right away, if a peer is strategic about sending
+    // sketches to us via different connections (requires attacker to occupy multiple outgoing connections).
+    if (msg_type == NetMsgType::SKETCH) {
+        if (peer->m_recon_state == nullptr) return;
+        if (peer->m_recon_state->m_outgoing_recon != Peer::ReconState::ReconPhase::INIT_REQUESTED) return;
+
+        std::vector<uint8_t> skdata;
+        vRecv >> skdata;
+
+        // Attempt to decode the received sketch with a local sketch.
+        if (skdata.size() / BYTES_PER_SKETCH_CAPACITY > MAX_SKETCH_CAPACITY) return;
```

Comment:
I think this is a protocol violation and leaves the link in a permanently broken reconciliation state, since a reconcildiff would never be sent. Perhaps we should disconnect the peer if this happens?

Reply:
Not really permanently broken, they still have a chance to send us a proper sketch. But yeah, either finalizing or disconnecting would work. Making it disconnect for now.